### DISCUSSION TITLE
Error validating: The relation field `tags` uses the scalar fields tagId. At least one of those fields is required. Hence the relation field must be required as well.

### DISCUSSION THREAD

**Discussion by NickSchmitt** [prisma/prisma#6555] [Q&A] [answered]
I am trying to do an explicit M:N join table.

I thought that this would be fixed in 2.21.0 due to [this thread](https://github.com/prisma/prisma/issues/6042) however I think I misunderstand my problem because it's still not working.

My first error is:
```
Error validating field `tags` in model `ProductsOnTags`: The relation field `tags` on Model `ProductsOnTags` is missing an opposite relation field on the model `Tag`. Either run `prisma format` or add it manually.
```

When I try to run prisma format, I get the following errors:

```
Error: Schema parsing
error: Error validating: The relation field `tags` uses the scalar fields tagId. At least one of those fields is required. Hence the relation field must be required as well.

error: Error validating: The relation field `products` uses the scalar fields productId. At least one of those fields is required. Hence the relation field must be required as well.
```

Here is my schema. Any thoughts on what could be the issue?

```prisma
model Product {
  id       Int      @id @default(autoincrement())
  title    String
  desc     String
  tags     Tag[]
}

model Tag {
  id       Int       @id @default(autoincrement())
  name     String
  products Product[]
}

model ProductsOnTags {
  id        Int       @id @default(autoincrement())
  tagId     Int
  tags      Tag[]     @relation(fields: [tagId], references: [id])
  productId Int
  products  Product[] @relation(fields: [productId], references: [id])
  @@unique([tagId, productId])
}
```

**Comment by ryands17** [comment, marked as answer]
@NickSchmitt 👋 
What you are creating is an [explicit many-to-many relation](https://www.prisma.io/docs/concepts/components/prisma-schema/relations#implicit-vs-explicit-many-to-many-relations). So your schema is indeed incorrect and should look like:

```prisma
model Product {
  id    Int              @id @default(autoincrement())
  title String
  desc  String
  tags  ProductsOnTags[]
}

model Tag {
  id       Int              @id @default(autoincrement())
  name     String
  products ProductsOnTags[]
}

model ProductsOnTags {
  id        Int      @id @default(autoincrement())
  product   Product? @relation(fields: [productId], references: [id])
  productId Int?
  tag       Tag?     @relation(fields: [tagId], references: [id])
  tagId     Int?

  @@unique([productId, tagId])
}
```

Have a look at the docs as well for a better idea on creating relations with Prisma :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
